pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -499,19 +499,27 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 			changes.CreateSandbox = false
 			return changes
 		}
+
+		// Get the containers to start, excluding the ones that succeeded if RestartPolicy is OnFailure.
+		// We should not create a sandbox for a Pod if there is no container to start.
```

Comment:
nit: move this comment to the corresponding if statement: `if len(containersToStart) == 0 {`

Reply:
done